src/data_member.rs

```diff
@@ -72,6 +72,108 @@ impl<T: Sized + Copy> DataMember<T> {
             _phantom: std::marker::PhantomData,
         }
     }
+
+    /// Create a new `DataMember` from a [`ProcessHandle`] and some number of (potentially negative) offsets.
+    /// You must remember to call [`try_into_process_handle`] on a [`Pid`] as sometimes the `Pid` can have
+    /// the same backing type as a [`ProcessHandle`], resulting in an error.
+    ///
+    /// [`try_into_process_handle`]: trait.TryIntoProcessHandle.html#tymethod.try_into_process_handle
+    /// [`ProcessHandle`]: type.ProcessHandle.html
+    /// [`Pid`]: type.Pid.htm
+    #[must_use]
+    #[allow(clippy::cast_sign_loss)]
+    pub fn new_offset_relative(handle: ProcessHandle, offsets: Vec<isize>) -> Self {
+        Self {
+            // Yes, we are casting to usize. This will not touch any bits, but due to 2s complement,
+            // we still get the correct result when adding offsets.
+            offsets: offsets.into_iter().map(|x| x as usize).collect(),
+            process: handle,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// Create a new `DataMember` from a [`ProcessHandle`], pointing to the memory address in the
+    /// remote process. Equivalent to `new_offset(handle, vec![addr])`. You must
+    /// remember to call [`try_into_process_handle`] on a [`Pid`] as sometimes the `Pid` can have
+    /// the same backing type as a [`ProcessHandle`], resulting in an error.
+    ///
+    /// [`try_into_process_handle`]: trait.TryIntoProcessHandle.html#tymethod.try_into_process_handle
+    /// [`ProcessHandle`]: type.ProcessHandle.html
+    /// [`Pid`]: type.Pid.html
+    #[must_use]
+    pub fn new_addr(handle: ProcessHandle, addr: usize) -> Self {
+        Self {
+            offsets: vec![addr],
+            process: handle,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// Create a new `DataMember` from a [`ProcessHandle`], pointing to the memory address in the
+    /// process, then following a bunch of pointers and offsets, which may be negative.
+    /// If you use CheatEngine and get a pointer of form "MyModule.dll + 0x12345678", plus a bunch
+    /// of offsets, then you want to put the base address of the module as `addr`, `0x12345678` as
+    /// the first offset, then any further offsets etc.
```

Comment:
Could you add a newline after this? I think that when it gets rendered into HTML, single linebreaks are removed.